classpath/java/io/BufferedInputStream.java

```diff
@@ -43,7 +43,12 @@ public int read() throws IOException {
 
   public int read(byte[] b, int offset, int length) throws IOException {
     int count = 0;
-
+    if (position >= limit) {
+	fill();
+      if (limit == -1) {
+        return -1;
+      }
+	}
```

Comment:
Copied this from the simple read() method, but i can easily change this too. Btw. The indentation was caused by that crazy windows editor i used. Will now correct it on Linux.